analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+                cancer_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                cancer_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                cancer_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## Hypertension
+        cov_bin_hypertension=(
+            (last_matching_event_clinical_snomed_before(
+                hypertension_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_clinical_ctv3_before(
+                hypertension_ctv3, index_date
+            ).exists_for_patient()) |
+            (last_matching_med_dmd_before(
+                hypertension_drugs_dmd, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                hypertension_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                hypertension_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## Diabetes # to add hba1c?
+        cov_bin_diabetes=(
+            (last_matching_event_clinical_snomed_before(
+                diabetes_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_med_dmd_before(
+                diabetes_drugs_dmd, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                diabetes_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                diabetes_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+      ## Pneumonia
+        cov_bin_history_pneumonia_snomed= (
+            last_matching_event_clinical_snomed_before(
+            pneumonia_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+      ## Asthma
+        cov_bin_history_asthma_snomed= (
+            last_matching_event_clinical_snomed_before(
+            asthma_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+      ## Pulmonary Fibrosis
+        cov_bin_history_pulmonary_fibrosis_snomed= (
+            last_matching_event_clinical_snomed_before(
+            pulmonary_fibrosis_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+        ## Ischaemic stroke
+        cov_bin_stroke_isch=(
+            (last_matching_event_clinical_snomed_before(
+                stroke_isch_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## All stroke (included the all-stroke codelists)
```

Comment:
I don't think all stroke is needed - I can't see it on the protocol and we usually use just ischaemic (expect for the cardiovascular specific project).

Reply:
Removed